src/services/ssdi.ts

```diff
@@ -1,197 +1,632 @@
+  defaultValues?: string[] | number[] | boolean[] | Record<string, any>[];
+  conditions?: string[];
 }
 
-// Caches to store the parsed query files and their metadata
-const queryFileCache: Map<string, QueryFile> = new Map();
-const queryDataCache: Map<string, CachedQuery> = new Map();
+interface Filters {
+  [key: string]: Filter;
+}
 
-// Helper function to read name, description, and default output name from a file
-const readNameAndDescriptionFromFile = (
-  filePath: string,
-): {
-  name: string;
-  description: string;
-  defaultOutputName: string;
-} => {
-  const fileContents = fs.readFileSync(filePath, 'utf8');
-  const nameMatch = fileContents.match(/@name:\s*(.*)/);
-  const descriptionMatch = fileContents.match(/@description:\s*(.*)/);
-  const defaultOutputNameMatch = fileContents.match(/@defaultOutputName:\s*(.*)/);
-  const fileName = path.basename(filePath, path.extname(filePath));
-  return {
-    name: nameMatch ? nameMatch[1].trim() : fileName,
-    description: descriptionMatch ? descriptionMatch[1].trim() : 'No description available',
-    defaultOutputName: defaultOutputNameMatch ? defaultOutputNameMatch[1].trim() : fileName,
-  };
+interface FilterValues {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [key: string]: string[] | number[] | boolean[] | Record<string, any>[];
+}
+
+interface CachedFilters {
+  filters: Filters;
+  supportsSorting: boolean;
+  supportsPagination: boolean;
+}
+
+// Base directory for file operations
+const BASE_DIRECTORY = path.resolve(process.cwd(), '/src/queries/');
+
+// Cache to store parsed JSON headers and queries
+const cache: Map<string, CachedFile> = new Map();
+
+// Sanitize the filename
+const sanitizeFilename = (filename: string): string => path.normalize(filename).replace(/[^a-zA-Z0-9-_]/g, '_');
+
+// Check if the user has access to a specific folders
+const checkFolderPermissions = async (
+  user,
+  scriptPath: string,
+) => (new Generic(user)).checkPermissions(
+  scriptPath,
+  ['dataRequests/ohs', 'dataRequests/internal'],
+  'ssdi_restricted',
+);
+
+// Safe file path resolution to prevent directory traversal attacks
+const safeResolvePath = (inputPath: string): string => {
+  const resolvedPath = path.resolve(BASE_DIRECTORY, inputPath);
+
+  if (!resolvedPath.startsWith(BASE_DIRECTORY)) {
+    throw new Error('Attempted to access a file outside the allowed directory.');
+  }
+
+  return resolvedPath;
 };
 
-// Helper function to list all query files with name and description
-const listQueryFiles = (directory: string): QueryFile[] => {
-  const files = fs.readdirSync(directory);
-  return files.map((file) => {
-    const filePath = path.join(directory, file);
-    if (queryFileCache.has(filePath)) {
-      return queryFileCache.get(filePath) as QueryFile;
+// Basic JSON validation function
+// TODO: make this more verbose validation
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const isValidJsonHeader = (json: any): boolean => (
```

Comment:
In this refactor, I’ve updated the `isValidJsonHeader` function to use **Zod** for runtime validation based on the `HeaderStructure` interface. This ensures that we only define the schema once and can validate at runtime.

### Steps Taken:

1. **Installed Zod and ts-to-zod**:
   - These packages allow for easy conversion of TypeScript interfaces into Zod schemas for validation.
   ~~~bash
   npm install zod ts-to-zod
   ~~~

2. **Converted TypeScript interfaces to Zod schemas**:
   - I used `ts-to-zod` to automatically convert the `HeaderFilter` and `HeaderStructure` interfaces into Zod schemas.

3. **Updated the `isValidJsonHeader` function**:
   - The function now uses Zod’s `.parse` method to validate JSON objects against the generated Zod schema. If validation fails, an error is caught, and `false` is returned.

### Example:

~~~typescript
import { z } from 'zod';
import { zodToSchema } from 'ts-to-zod';

// Define your existing TypeScript interfaces (HeaderFilter, HeaderStructure)

// Generate Zod schemas
const HeaderFilterSchema = zodToSchema<HeaderFilter>();
const HeaderStructureSchema = zodToSchema<HeaderStructure>();

// Refactor the function
const isValidJsonHeader = (json: unknown): boolean => {
  try {
    HeaderStructureSchema.parse(json); // Validate JSON
    return true;
  } catch (e) {
    console.error(e.errors); // Handle validation errors
    return false;
  }
};
~~~

### Benefits:
- **Single Source of Truth**: The schema is derived from the existing TypeScript interface, which avoids duplicating logic for validation.
- **Robust Runtime Validation**: Zod provides detailed validation, including the ability to handle complex structures.

### Summary:
With this refactor, the validation logic for `isValidJsonHeader` is simplified and made more maintainable, as it relies on the Zod schema generated from the existing TypeScript interfaces. This allows both compile-time and runtime validation, ensuring that the JSON objects conform to the defined structure.

Reply:
I'm not sure that this is a problem we need to solve for the QA dash MVP, but putting that aside we already have a schema validation library installed, and it would be better to a) leverage that if possible or b) convert usage of Joi to Zod elsewhere in the repo

Reply:
zod has been removed